dev/VSIX/Extension/Cpp/Dev17/WizardImplementationCpp.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License
+
+using EnvDTE;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TemplateWizard;
+using NuGet.VisualStudio;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using WindowsAppSDK.Cpp.Extension;
+
+namespace WindowsAppSDK.TemplateUtilities.Cpp
+{
+    public class NuGetPackageInstaller : IWizard
+    {
+        private Project _project;
+        private IComponentModel _componentModel;
+        public void RunStarted(object automationObject, Dictionary<string, string> replacementsDictionary, WizardRunKind runKind, object[] customParams)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            _componentModel = (IComponentModel)ServiceProvider.GlobalProvider.GetService(typeof(SComponentModel));
+        }
+        public void ProjectFinishedGenerating(Project project)
+        {
+            _project = project;
+
+            // Ensure we're on the main thread, as required by the ProjectFinishedGenerating method and DTE operations
+            ThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
+                await InstallNuGetPackagesAsync(_project);
+            });
+        }
+        private async Task InstallNuGetPackagesAsync(Project project)
+        {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            foreach (var packageId in NuGetPackageList.Packages)
```

Comment:
with multiple support now, can we make Microsoft.Windows.SDK.BuildTools dynamic in the C# templates?

also, a bit of cleanup: 
```
      <CustomParameter Name="$WindowsAppSDKNupkgVersion$" Value="FIXME-Verify-Directory.Build.Targets-XmlPoke-Queries" />
      <CustomParameter Name="$WindowsSDKBuildToolsNupkgVersion$" Value="FIXME-Verify-Directory.Build.Targets-XmlPoke-Queries" />
```
etc.

Reply:
I guess this is a separate implementation from C#?  But would still like to see multiple support added there.

Reply:
Yes, it's a separate implementation that allows us to install multiple packages compared to the existing C# method that was better suited for only one package.

Reply:
C# would also benefit from multi-package support (BuildTools)